### PR TITLE
REGR: fix DataFrame.agg case with list-like return value

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref https://github.com/pandas-dev/pandas/pull/29587

cc @jbrockmendel

### CODE REVIEW

File: pandas/tests/frame/test_apply.py [hunk 1]
```diff
@@ -1259,6 +1259,23 @@ def test_non_callable_aggregates(self):
 
         assert result == expected
 
+    def test_agg_listlike_result(self):
+        # GH-29587 user defined function returning list-likes
```
Comment (jbrockmendel) [inline pandas/tests/frame/test_apply.py]: can we say something snide about this user?  i dislike this hypothetical user.

**Comment by jbrockmendel** [comment]
LGTM

**Review by WillAyd — APPROVED** [review]
lgtm whenever green

**Comment by jorisvandenbossche** [comment]
Red travis ci was the failing database

**Pull request merged**